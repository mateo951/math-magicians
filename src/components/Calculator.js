import React from 'react';

const Items = () => <h1>Current Quantity</h1>;

export default Items;

// class Calculator extends React.Component {
//   render() {
//     return <Items quantity={ this.props.quantity }/>;
//   }
// }

// export default Calculator;