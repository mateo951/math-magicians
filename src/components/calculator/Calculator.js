import React, { useState } from 'react';
import Row from './Row';
import calculate from '../../logic/calculate/calculate';
import style from './calc.module.css';

const Calculator = () => {
  const [obj, setObj] = useState({ total: 0, next: null, operation: null });
  const { total, next, operation } = obj;

  const handleClick = (e) => {
    setObj((prevState) => calculate(prevState, e.target.value));
  };

  const displayTotal = () => {
    if ((total === 0) && next === null) {
      return 0;
    }
    if (total !== 0 && total !== null) {
      return total;
    }
    return '';
  };
  return (
    <div>
      <h3 className={style.title}>Lets do some math!</h3>
      <div className="calcContainer">
        <input type="text" id="result" value={displayTotal() + (operation || '') + (next || '')} disabled />
        {[['AC', '+/-', '%', 'รท'], ['7', '8', '9', 'x'], ['4', '5', '6', '-'], ['1', '2', '3', '+'], ['0', '.', '=']].map((subArray) => (
          <div key={subArray[0]}>
            <Row value={subArray[0]} className={subArray[0] === '0' ? 'cero' : ''} handleClick={handleClick} />
            <Row value={subArray[1]} handleClick={handleClick} />
            <Row value={subArray[2]} handleClick={handleClick} />
            {subArray.length === 4 && <Row value={subArray[3]} className={`${subArray.length === 4 ? 'operators' : ''}`} handleClick={handleClick} />}
          </div>
        ))}
      </div>
    </div>
  );
};

export default Calculator;
