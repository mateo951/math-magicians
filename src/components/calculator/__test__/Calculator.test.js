import React from 'react';
import ReactDOM from 'react-dom';
import {
  render, screen, cleanup, fireEvent,
} from '@testing-library/react';
import Calculator from '../Calculator';

let div = null;
beforeEach(() => {
  div = document.createElement('div');
  document.body.appendChild(div);
  render(<Calculator />);
});
afterEach(cleanup);

describe('Test rendering', () => {
  it('renders with out crashing', () => {
    ReactDOM.render(<Calculator />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders text correctly', () => {
    const calculator = screen.getByText('Lets do some math!');
    expect(calculator).toMatchSnapshot();
  });

  it('renders calculator operations', () => {
    const operations = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
    operations.forEach((operation) => {
      expect(screen.getByText(operation)).toBeInTheDocument();
    });
  });

  it('5 + 4', () => {
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('='));

    expect(screen.getByText('9')).toMatchSnapshot();
  });
});
