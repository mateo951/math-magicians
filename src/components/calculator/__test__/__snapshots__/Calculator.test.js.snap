// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering 5 + 4 1`] = `
<input
  class="calcItem "
  type="button"
  value="9"
/>
`;

exports[`Test rendering renders text correctly 1`] = `
<h3
  class="title"
>
  Lets do some math!
</h3>
`;
